### PR TITLE
[receiver/oracledb] Fix incorrect values for a couple of metrics

### LINKED ISSUE #31451: [receiver/oracledb] discrepancy between the metric values in DBA_TABLESPACE_USAGE_METRICS and the ones calculated by the receiver
### Component(s)

receiver/oracledb

## Description
The oracledb receiver uses the below [queries] respectively (https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/oracledbreceiver/scraper.go#L41-L42) to set values for `oracledb.tablespace_size.limit` and `oracledb.tablespace_size.usage` in bytes
```
	tablespaceUsageSQL      = "select TABLESPACE_NAME, BYTES from DBA_DATA_FILES"
	tablespaceMaxSpaceSQL   = "select TABLESPACE_NAME, (BLOCK_SIZE*MAX_EXTENTS) AS VALUE FROM DBA_TABLESPACES"
```
Our backend uses those metrics to calculate the % used tablespace `(oracledb.tablespace_size.usage/oracledb.tablespace_size.limit)*100`

However, when we look at the values in table `DBA_TABLESPACE_USAGE_METRICS` https://docs.oracle.com/en/database/oracle/oracle-database/19/refrn/DBA_TABLESPACE_USAGE_METRICS.html#GUID-FE479528-BB37-4B55-92CF-9EC19EDF4F46 , the percentage used and the table size/used space, although in data blocks, still do not match the receiver's datapoints when converted. 

## Steps to Reproduce
enable [the oracledb receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/oracledbreceiver) and compare `(oracledb.tablespace_size.usage/oracledb.tablespace_size.limit)*100` to `USED_PERCENT` from `DBA_TABLESPACE_USAGE_METRICS`

## Expected Result
Both values should match

## Actual Result
Different values

### Collector version

0.93.0

### PR DESCRIPTION
**Pull request by crobert-1**
**Description:** <Describe what has changed.>

Values were being scraped incorrectly for the metrics `oracledb.tablespace_size.limit` and `oracledb.tablespace_size.usage`. The changes these metrics to be scraped from the [`DBA_TABLESPACE_USAGE_METRICS`](https://docs.oracle.com/en/database/oracle/oracle-database/19/refrn/DBA_TABLESPACE_USAGE_METRICS.html#GUID-FE479528-BB37-4B55-92CF-9EC19EDF4F46) table. This results in a slight loss of granularity in these metrics, as values will always be in multiples of the respective tablespace's block size, but I think the clarity and simplicity is worth the trade off.

Note: The value of the usage metric was generally close to the expected value, but the limit was being calculated as potential theoretical capacity, unbound by server capacity. For example, in testing in a docker container on my local machine, limit was set to **17TB**. This doesn't line up with user expectations.

**Link to tracking Issue:** <Issue number if applicable>
Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31451

**Testing:** <Describe what testing was performed and which tests were added.>
Updated existing tests, added a couple new ones.

Also, the original issue filed was comparing `DBA_TABLESPACE_USAGE_METRICS` output for percent used to what we got from `usage/limit * 100`. Here's the local testing outputs compared to show they now line up.
```
2024-03-27T16:31:57.938-0700    info    oracledbreceiver/scraper.go:285 DBA_TABLESPACE_USAGE_METRICS: Tablespace name: SYSTEM, used space: 111288, tablespace size: 3518587, percent used: 3.16286054600895188892586711654422641816    {"kind": "receiver", "name": "oracledb", "data_type": "metrics"}
```

```
Metric #20
Descriptor:
     -> Name: oracledb.tablespace_size.usage
     -> Description: Used tablespace in bytes.
     -> Unit: By
     -> DataType: Gauge
NumberDataPoints #0
Data point attributes:
     -> tablespace_name: Str(SYSTEM)
StartTimestamp: 2024-03-27 23:31:56.873576 +0000 UTC
Timestamp: 2024-03-27 23:32:12.523295 +0000 UTC
Value: 911671296
```

```
Metric #19
Descriptor:
     -> Name: oracledb.tablespace_size.limit
     -> Description: Maximum size of tablespace in bytes, -1 if unlimited.
     -> Unit: By
     -> DataType: Gauge
NumberDataPoints #0
Data point attributes:
     -> tablespace_name: Str(SYSTEM)
StartTimestamp: 2024-03-27 23:31:56.873576 +0000 UTC
Timestamp: 2024-03-27 23:32:12.523295 +0000 UTC
Value: 28824264704
```
Doing the same calculation, we get:
```
(911671296 / 28824264704) * 100 = ~3.16%
```

### CODE REVIEW

File: receiver/oracledbreceiver/metadata.yaml [hunk 1]
```diff
@@ -233,7 +233,6 @@ metrics:
     enabled: true
     gauge:
       value_type: int
-      input_type: string
```
Comment (crobert-1) [inline receiver/oracledbreceiver/metadata.yaml:236]: The logic now requires two DB values to be multiplied by each other. This means we have to manually convert both values to `int64`, multiply them, then use that result to set the value here. Converting back to string just to convert to `int64` again is unnecessary, so I've removed this option.

File: receiver/oracledbreceiver/scraper.go [hunk 2]
```diff
@@ -38,8 +38,11 @@ const (
 	consistentGets          = "consistent gets"
 	sessionCountSQL         = "select status, type, count(*) as VALUE FROM v$session GROUP BY status, type"
 	systemResourceLimitsSQL = "select RESOURCE_NAME, CURRENT_UTILIZATION, LIMIT_VALUE, CASE WHEN TRIM(INITIAL_ALLOCATION) LIKE 'UNLIMITED' THEN '-1' ELSE TRIM(INITIAL_ALLOCATION) END as INITIAL_ALLOCATION, CASE WHEN TRIM(LIMIT_VALUE) LIKE 'UNLIMITED' THEN '-1' ELSE TRIM(LIMIT_VALUE) END as LIMIT_VALUE from v$resource_limit"
-	tablespaceUsageSQL      = "select TABLESPACE_NAME, BYTES from DBA_DATA_FILES"
-	tablespaceMaxSpaceSQL   = "select TABLESPACE_NAME, (BLOCK_SIZE*MAX_EXTENTS) AS VALUE FROM DBA_TABLESPACES"
+	tablespaceUsageSQL      = `
```
Comment (crobert-1) [inline receiver/oracledbreceiver/scraper.go:41]: Feedback on indentation here would be helpful, not sure what's the most clear here.

**Review by atoulme — APPROVED** [review]

File: receiver/oracledbreceiver/scraper.go (hunk 2 above)
Comment (atoulme) [reply receiver/oracledbreceiver/scraper.go:41]: It was readable enough. I had a bit of hesitation that the query itself is 2 separate subqueries, joined via a where statement with a bigger query. It feels like an inner join could do the job, but my SQL is rusty and this is an improvement over the broken functionality. Maybe something to explore if it becomes a performance flare up, which I doubt.

Comment (crobert-1) [reply receiver/oracledbreceiver/scraper.go:41]: Fair point. I added a benchmark test locally to compare this query vs. inner join, here are the results.
Inner join:
```
BenchmarkSQLUsageMetrics-16    	     150	   6881141 ns/op
BenchmarkSQLUsageMetrics-16    	     180	   6503225 ns/op
BenchmarkSQLUsageMetrics-16    	     172	   6594906 ns/op
BenchmarkSQLUsageMetrics-16    	     154	   6556032 ns/op
```
Two separate queries:
```
BenchmarkSQLUsageMetrics-16    	     132	   8800545 ns/op
BenchmarkSQLUsageMetrics-16    	     122	   8808408 ns/op
BenchmarkSQLUsageMetrics-16    	     118	   9132146 ns/op
BenchmarkSQLUsageMetrics-16    	     120	   8762194 ns/op
```
I'll update to use the inner join 👍

**Pull request merged**